packages/seo-store/src/editor/selectors/selectFormattedDate.js

```diff
@@ -0,0 +1,43 @@
+import { get } from "lodash";
+
+/**
+ * Formats the date in a way that is expected
+ * in the Google Preview and SEO analysis.
+ *
+ * @param {string} dateString The date in ISO 8601 format.
+ * @param {string} language The language in which to format the date.
+ *
+ * @returns {string} The formatted date.
+ */
+function formatDate( dateString, language ) {
+	const date = new Date( dateString );
+
+	const month = new Intl.DateTimeFormat( language, { month: "short" } ).format( date );
+	const day = new Intl.DateTimeFormat( language, { day: "numeric" } ).format( date );
+	const year = new Intl.DateTimeFormat( language, { year: "numeric" } ).format( date );
```

Comment:
Using the language here (from the store) means that the formatted date is in the user's language, which should fix [this issue](https://yoast.atlassian.net/browse/LINGO-1472).